Design/nexusgeometrysaving.md

```diff
@@ -0,0 +1,74 @@
+## Overview
+
+Version 4.0 of Mantid saw the introduction of the Nexus Geometry Format [updates](https://docs.mantidproject.org/v4.0.0/release/v4.0.0/framework.html#nexus-geometry-loading) into Mantid. This has gone well with, many of our tests showing speed improvements upon loading for a range of equivalant instruments (Rectangular Detectors excluded) and no loss of functionality or performance for the existing routes.
+
+ESS intend to support the new geometry format for intermediate files because:
+
+1. The format has been demonstrated to suit the representation of neutron instrument geometries well
+1. It is relatively low-cost for the ESS to adopt as part of an intermediate format.
+1. It gives a common format for geometry for Raw and Processed files
+1. It is complete and accurate. The current approach bypasses serialization.
+1. It is not application specific, geometries are useable in downstream applications more easily.
+
+This is a follow up to the Working Group Meeting 20/06/2019. We are aware that ISIS do not intend to make use of the new format. The positions of the SNS and ILL were not represented at that time.
+
+We are currently running a small project involving a student to provide a saving mechanism within the existing the lib [NexusGeometry](https://github.com/mantidproject/mantid/tree/master/Framework/NexusGeometry) to provide a high-level function to support the saving.
+
+**This document is for discussion of the best way to make these features available within Mantid**
+
+## 1. Implicit Replace instrument_xml in `SaveNexusProcessed`
+
+This represents an implicit change in behaviour.
+
+Current mechanism [implemented here](https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument.cpp#L1010), would be replaced (see notes below) to use the new method in `NexusGeometry`. This would mean that ALL new processed nexus files contained the full geometry rather than the xml string. Aligned with that, we would also remove the instrument geometry specific parameters i.e. the things representing overloads to the correct position.
```

Comment:
Can I check what happens with non-instrument parameters set in the IDFs? I assume these are still saved/loaded?

Reply:
Yes, they must be saved still. Though we are not discussing instrument 2.0 here (i don't want to get the Nexus geometry aspects confused with this) it uses a similar process to make parameters that allow current processed nexus files to be writen (Instrument 2.0 also has absolute positions/rotation for all items) see [here](https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument.cpp#L1010)